src/main/java/no/digipost/api/client/MessageSender.java

```diff
@@ -83,28 +87,42 @@ public MessageDelivery sendMultipartMessage(Message message, Map<String, Documen
 		Map<Document, InputStream> documentInputStream = encryptionAndInputStream.documentsAndInputstream;
 		Message singleChannelMessage = encryptionAndInputStream.getSingleChannelMessage();
 
-		try (MultiPart multiPart = new MultiPart()) {
-			BodyPart messageBodyPart = new BodyPart(singleChannelMessage, MediaType.valueOf(MediaTypes.DIGIPOST_MEDIA_TYPE_V6));
-			ContentDisposition messagePart = ContentDisposition.type("attachment").fileName("message").build();
-			messageBodyPart.setContentDisposition(messagePart);
-			multiPart.bodyPart(messageBodyPart);
-
+		try {
 			Map<Document, InputStream> preparedDocuments = documentsPreparer.prepare(
 					documentInputStream, singleChannelMessage, encrypter, Apply.partially(resolvePdfValidationSettings).of(singleChannelMessage));
+
+			ByteArrayOutputStream bao = new ByteArrayOutputStream();
+			JAXB.marshal(singleChannelMessage, bao);
+			ByteArrayBody attachment = new ByteArrayBody(bao.toByteArray(),
+					ContentType.create(MediaTypes.DIGIPOST_MEDIA_TYPE_V6),"message");
+
+			MultipartEntityBuilder multipartEntity = MultipartEntityBuilder.create().setMode(HttpMultipartMode.STRICT).setMimeSubtype("mixed")
+					.addPart(FormBodyPartBuilder.create("message", attachment).addField("Content-Disposition", "attachment;" + " filename=\"message\"").build());
```

Comment:
Litt keitete å lese den her linja. Hva med et linjeskift for hvert metodekall her?